fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    fetch('/api/todos')
+      .then(response => response.json())
+      .then(data => this.setState({ todos: data }));
+  }
+
+  addTodo(newTodo) {
+
+    const requestOptions = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ task: newTodo.task })
+    };
+
+    fetch('/api/todos', requestOptions)
+      .then(response => response.json())
+      .then(data => this.setState({ todos: this.state.todos.concat(data) })
+      );
+  }
+
+  toggleCompleted(todoId) {
+    const isCompletedToggleState = !this.state.todos[todoId - 1].isCompleted;
```

Comment:
You can't use the id as an index here either, you can't expect ids and indexes to always line up. You need to use `.find` or `.findIndex` to find the correct item in the array as you did below.